synapse/rest/client/v1/room.py

```diff
@@ -458,6 +458,9 @@ async def on_POST(self, request, room_id):
                 "state_key": state_event["state_key"],
             }
 
+            # Mark all events as historical
+            event_dict["content"][EventContentFields.MSC2716_HISTORICAL] = True
```

Comment:
This is just the optional historical decoration field. But would be good to have all of the events include it.